Client/src/Pages/Settings Pages/StudentSettings/ProfileSettings/ConfirmImageDialog.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import {Dialog, DialogTitle, DialogContent, DialogActions, Button, Typography, CircularProgress} from '@material-ui/core'
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+
+export default class ConfirmImageModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            open: false,
+            picture: null,
+            loading: false
+        }
+    }
+
+    toggleOpen = () => {
+        this.setState({ open: !this.state.open });
+    }
+
+    async handleProfileChange(file) {
+        this.setState({
+            open: true, // Better to let the user know that image is being processed.
+            loading: true
+        });
+
+        file = await this.convertToJPEG(file);
```

Comment:
What happens in the UI when this function throws an error?

Reply:
I am confused about what scenarios could we expect the function to throw an error. We are restricting the file extension in the FilePicker UI component, and this method just uses a canvas to draw an image and then to save that canvas in a different extension. Could you please elaborate in what scenarios I could expect an error so I could handle it properly?